src/Sentry/DynamicSamplingContext.cs

```diff
@@ -111,6 +123,27 @@ private DynamicSamplingContext(
             return null;
         }
 
+        // See https://develop.sentry.dev/sdk/telemetry/traces/#propagated-random-value
+        if (items.TryGetValue("sample_rand", out var sampleRand))
```

Comment:
odd that we validate this to be within a range as a float but then we make it a string then try to parse it back out.
Can we rely on the original value and avoid the parsing instead?

Reply:
I'm not sure I follow. It may or may not be present as a string in the baggage header. If it's present then here we try to cast it to a float and ensure it's in the correct range. 

If it's present but malformed then we don't create the DSC from the baggage header.